src/status_im/subs/wallet/networks.cljs

```diff
@@ -45,38 +84,99 @@
    (-> networks (get chain-id) :network-name)))
 
 (re-frame/reg-sub
- :wallet/selected-network-details
- :<- [:wallet/network-details]
- :<- [:wallet/selected-networks]
- (fn [[network-details selected-networks]]
-   (filter
-    #(contains? selected-networks (:network-name %))
-    network-details)))
+ :wallet/active-chain-ids
+ :<- [:wallet/networks]
+ (fn [networks]
+   (networks/get-active-chain-ids networks)))
 
 (re-frame/reg-sub
- :wallet/network-values
- :<- [:wallet/networks-by-id]
- :<- [:wallet/wallet-send]
- (fn [[networks {:keys [from-values-by-chain to-values-by-chain token-display-name token] :as send-data}]
-      [_ to-values?]]
-   (let [network-values (if to-values? to-values-by-chain from-values-by-chain)
-         token-symbol   (or token-display-name
-                            (-> send-data :token :symbol))
-         token-decimals (:decimals token)]
-     (reduce-kv
-      (fn [acc chain-id amount]
-        (let [network-name (-> networks (get chain-id) :network-name)
-              amount-fixed (number/to-fixed (money/->bignumber amount) token-decimals)]
-          (merge acc (network-utils/network-summary network-name token-symbol amount-fixed))))
-      {}
-      network-values))))
-
-(re-frame/reg-sub
- :wallet/send-selected-network
- :<- [:wallet/networks-by-id]
- :<- [:wallet/wallet-send]
- (fn [[networks {:keys [to-values-by-chain]}]]
-   (->> to-values-by-chain
-        keys
-        first
-        (get networks))))
+ :wallet/active-networks
+ :<- [:wallet/networks]
+ (fn [networks]
+   (networks/get-active-networks networks)))
+
+(re-frame/reg-sub
+ :wallet/filtered-networks
+ :<- [:wallet/active-networks]
+ :<- [:wallet/network-filter]
+ (fn [[active-networks network-filter]]
+   (networks/get-filtered-networks active-networks network-filter)))
+
+(re-frame/reg-sub
+ :wallet/filtered-chain-ids
+ :<- [:wallet/active-networks]
+ :<- [:wallet/network-filter]
+ (fn [[active-networks network-filter]]
+   (networks/get-filtered-chain-ids active-networks network-filter)))
+
+(re-frame/reg-sub
+ :wallet/network-filter-toggled?
+ :<- [:wallet/filtered-chain-ids]
+ (fn [filtered-chain-ids [_ chain-id]]
+   (contains? filtered-chain-ids chain-id)))
+
+(re-frame/reg-sub
+ :wallet/disable-network-filter?
+ :<- [:wallet/filtered-chain-ids]
+ (fn [filtered-chain-ids [_ chain-id]]
+   (and (= 1 (count filtered-chain-ids))
+        (contains? filtered-chain-ids chain-id))))
+
+(re-frame/reg-sub
+ :wallet/layer-1-networks
+ :<- [:wallet/active-networks]
+ (fn [networks]
+   (networks/get-networks-for-layer networks 1)))
+
+(re-frame/reg-sub
+ :wallet/eth-mainnet-network
+ :<- [:wallet/layer-1-networks]
+ (fn [networks]
+   (first networks)))
+
+(re-frame/reg-sub
+ :wallet/layer-2-networks
+ :<- [:wallet/active-networks]
+ (fn [networks]
+   (networks/get-networks-for-layer networks 2)))
```

Comment:
Highlighting that we will need to remove the layer categorisation in UI soon, and this subs probably will be cleaned up.